pkg/util/tracing/tracer_test.go

```diff
@@ -165,15 +165,6 @@ func TestTracerRecording(t *testing.T) {
 	s1.Recordf("x=%d", 100)
 	require.Nil(t, s1.GetRecording(RecordingStructured))
 
-	// The child Span, now finished, will drop future recordings.
```

Comment:
Does recording into finished spans error out or drop things silently? Are we worried about unintended use after finish bugs?

Reply:
At the moment, such recordings get dropped silently.
My plan for the near future is to make span use-after-Finish crash in unit tests and roachtests, but otherwise probably tolerate it in production at least for the time being. I've been fixing a bunch of use-after-Finishes, but there's probably more. Once we start re-using spans (through a sync.Pool), span-use-after-Finish can only be detected if the finished span is used before it is reallocated. When it's used after re-allocation, that will result in trace corruption (e.g. you'll have a span get messages from two unrelated operations).